plugins/MidiImport/MidiImport.cpp

```diff
@@ -315,19 +316,21 @@ bool MidiImport::readSMF( TrackContainer* tc )
 
 	pd.setValue( 0 );
 
-	std::stringstream stream;
-	QByteArray arr = readAllData();
-	stream.str(std::string(arr.constData(), arr.size()));
-
-	Alg_seq_ptr seq = new Alg_seq(stream, true);
+	std::ifstream iFile(file().fileName().toStdString().c_str(), std::ios::in | std::ios::binary);
+	Alg_seq_ptr seq = new Alg_seq(iFile, true);
+	iFile.close();
```

Comment:
Please don't change this unless the updated portsmf complains. I changed it in #4401 due to some issues on Windows related to file name encoding.

Reply:
I missed the comment above. I think you can still replace `std::stringstream` with `std::istringstream` without further changes.

Reply:
I tried, and during runtime it kept complaining that it couldn't find a valid constructor. I'll give it a shot again though and see if I can get it working.

Reply:
Somehow it works now. I don't know what happened before. 😩

While looking at it I figured out the `ifstream` way actually didn't need to add the `.c_str()` part for the filename. Do you think using a `std::string` for the filename would handle the encoding issues you found? I think using an `ifstream` is a little cleaner code, and less converting from `QByteArray` to `std::string` when we can just pass the file directly.

Reply:
On Windows, the standard C++ I/O can't handle file names outside the local code page, that's why I used `QFile`. If we can wrap it with a better way, it will be great.

Reply:
Okay I've cleaned up the code to be simpler and more readable, and still use the QFile method.